### PR TITLE
Test Events

### PR DESCRIPTION
**Pull request by justkawal**
## Any follow ups?
Please review PR: #141, before reviewing this PR.

## Description
* Adds events testing

## Testing Plan 
`dart test`

### CODE REVIEW

File: packages/substrate_metadata/test/events_test.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    //
+    // Populating with the metadata for block-numbers available for this chain....
+    chain.initSpecVersionFromFile('../../chain/polkadot/versions.jsonl');
+
+    final List<RawBlockEvents> rawBlocksList =
+        getEvents('../../chain/polkadot/events.jsonl');
+
+    //
+    // Looping through every block
+    for (var originalEvent in rawBlocksList) {
```
Comment (gabrielokura) [inline packages/substrate_metadata/test/events_test.dart:51]: these tests don't follow last bruno's tips.

He said we should avoid loops and must create readable tests titles

Comment (justkawal) [reply packages/substrate_metadata/test/events_test.dart:51]: I have a call with him, and for this case scenario we're going with it by like these.

Comment (justkawal) [reply packages/substrate_metadata/test/events_test.dart:51]: Its practically not possible, to search a sweet block which would be good enough to test. So, can't cherry pick any one.

**Comment by gabrielokura** [comment]
I see we are testing events, but what is the meaning of these tests? How they protect us against bugs in decode && encode methods?

I mean, shouldn't we test these two different methods separately?

**Comment by justkawal** [comment]
> I see we are testing events, but what is the meaning of these tests? How they protect us against bugs in decode && encode methods?
> 
> shouldn't we test these two different methods separately?

When you decode a value with use of the metadata definition only and then encode it back and get the same result as the original value then it confirms that the code understands the inputed value, as resulting in E2E testing....

Again, I'm not saying the code is 100% sweet and tight, but we will get it in future with `unit-tests.`

Future task: `unit-test` @gabrielokura .

**Comment by gabrielokura** [comment]
> 

understood!

**Comment by gabrielokura** [comment]
> > I see we are testing events, but what is the meaning of these tests? How they protect us against bugs in decode && encode methods?
> > shouldn't we test these two different methods separately?
> 
> When you decode a value with use of the metadata definition knowledge only and then encode it back and gets the same result as the original value then it confirms that the code understands the inputed value with the use of provided definitions or schema or metadata, as resulting in E2E testing....
> 
> Again, I'm not saying the code is 100% sweet and tight, but we will get it in future with `unit-tests.`
> 
> Future task: `unit-test` @gabrielokura .

understood!

File: packages/substrate_metadata/test/events_test.dart [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
```
Comment (pastre) [inline packages/substrate_metadata/test/events_test.dart:51]: Please create an issue to track and refactor this

**Review by pastre — CHANGES_REQUESTED** [review]
Need to track bad tests

File: packages/substrate_metadata/test/events_test.dart (hunk 2 above)
Comment (justkawal) [reply packages/substrate_metadata/test/events_test.dart:51]: Tracked with #150.

**Review by leonardocustodio — APPROVED** [review]

**Review by pastre — APPROVED** [review]

**Pull request merged**